tests/returntypes/test_autograd_qnode_new.py

```diff
@@ -0,0 +1,1942 @@
+            pytest.skip("Test only supports parameter-shift")
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qnode(dev, interface="autograd", diff_method="parameter-shift")
+        def circuit(x, y):
+            qml.RX(x[0], wires=0)
+            qml.Rot(*x[1:], wires=0)
+            qml.RY(y[0], wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        x = np.array([0.1, 0.2, 0.3, 0.4], requires_grad=False)
+        y = np.array([0.5], requires_grad=True)
+
+        param_data = []
+
+        def mock_apply(*args, **kwargs):
+            for op in args[0]:
+                param_data.extend(op.data.copy())
+
+        mocker.patch.object(dev, "apply", side_effect=mock_apply)
+        circuit(x, y)
+        assert param_data == [0.1, 0.2, 0.3, 0.4, 0.5]
+        assert not any(isinstance(p, np.tensor) for p in param_data)
+
+        # test the jacobian works correctly
+        param_data = []
+        qml.grad(circuit)(x, y)
+        assert param_data == [
+            0.1,
+            0.2,
+            0.3,
+            0.4,
+            0.5,
+            0.1,
+            0.2,
+            0.3,
+            0.4,
+            0.5 + np.pi / 2,
+            0.1,
+            0.2,
+            0.3,
+            0.4,
+            0.5 - np.pi / 2,
+        ]
+        assert not any(isinstance(p, np.tensor) for p in param_data)
+
+    def test_execution_no_interface(self, dev_name, diff_method, mode):
+        """Test execution works without an interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, interface=None)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = np.array(0.1, requires_grad=True)
+
+        res = circuit(a)
+
+        assert circuit.qtape.interface == None
+
+        # without the interface, the QNode simply returns a scalar array
+        assert isinstance(res, np.ndarray)
+        assert res.shape == tuple()
+
+        # gradients should cause an error
+        with pytest.raises(TypeError, match="must be real number, not ArrayBox"):
+            qml.grad(circuit)(a)
+
+    def test_execution_with_interface(self, dev_name, diff_method, mode):
+        """Test execution works with the interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = np.array(0.1, requires_grad=True)
+        assert circuit.interface == "autograd"
+        # gradients should work
+        grad = qml.grad(circuit)(a)
+
+        assert isinstance(grad, float)
```

Comment:
Should this be the case?
```suggestion
        assert isinstance(grad, np.ndarray)
```

Reply:
Same comment as before the numpy.array with shape (,) is automatically converted to float by autograd. I could not solve the issue.